modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/cdc/CdcCommand.java

```diff
@@ -0,0 +1,134 @@
+import org.apache.ignite.internal.commandline.Command;
+import org.apache.ignite.internal.commandline.CommandArgIterator;
+import org.apache.ignite.internal.commandline.CommandLogger;
+import org.apache.ignite.internal.visor.VisorTaskArgument;
+import org.apache.ignite.internal.visor.cdc.VisorCdcDeleteLostSegmentsTask;
+
+import static org.apache.ignite.internal.commandline.CommandList.CDC;
+import static org.apache.ignite.internal.commandline.CommandLogger.optional;
+import static org.apache.ignite.internal.commandline.CommonArgParser.CMD_AUTO_CONFIRMATION;
+import static org.apache.ignite.internal.commandline.TaskExecutor.executeTaskByNameOnNode;
+
+/**
+ * CDC command.
+ */
+public class CdcCommand extends AbstractCommand<String> {
+    /** Command to delete lost segment links. */
+    public static final String DELETE_LOST_SEGMENT_LINKS = "delete_lost_segment_links";
+
+    /** */
+    public static final String NODE_ID = "--node-id";
+
+    /** Node ID. */
+    private UUID nodeId;
+
+    /** {@inheritDoc} */
+    @Override public Object execute(GridClientConfiguration clientCfg, IgniteLogger log) throws Exception {
+        try (GridClient client = Command.startClient(clientCfg)) {
+            executeTaskByNameOnNode(
+                client,
+                VisorCdcDeleteLostSegmentsTask.class.getName(),
+                null,
+                nodeId,
+                clientCfg
+            );
+
+            Collection<UUID> nodeIds = nodeId != null ? Collections.singletonList(nodeId) :
+                client.compute().nodes(node -> !node.isClient()).stream().map(GridClientNode::nodeId)
+                    .collect(Collectors.toSet());
+
+            client.compute().execute(VisorCdcDeleteLostSegmentsTask.class.getName(),
+                new VisorTaskArgument<>(nodeIds, false));
+
+            String res = "Lost segment CDC links successfully removed.";
+
+            log.info(res);
+
+            return res;
+        }
+        catch (Throwable e) {
+            log.error("Failed to perform operation.");
+            log.error(CommandLogger.errorMessage(e));
+
+            throw e;
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override public void parseArguments(CommandArgIterator argIter) {
+        nodeId = null;
+
+        String cmd = argIter.nextArg("Expected command: " + DELETE_LOST_SEGMENT_LINKS);
+
+        if (!DELETE_LOST_SEGMENT_LINKS.equalsIgnoreCase(cmd))
+            throw new IllegalArgumentException("Unexpected command: " + cmd);
+
+        while (argIter.hasNextSubArg()) {
+            String opt = argIter.nextArg("Failed to read command argument.");
+
+            if (NODE_ID.equalsIgnoreCase(opt))
+                nodeId = argIter.nextUuidArg(NODE_ID);
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override public String confirmationPrompt() {
+        return "Warning: the command will delete lost segment CDC links. Cache events from these segments will be lost.";
```

Comment:
The command will fix WAL segments gap in case CDC link creation was stopped by distributed property or excess of maximum CDC directory size. Gap will be fixed by deletion of WAL segments previous to the gap.

Reply:
Fixed